xilem_core/src/deferred.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use core::{fmt::Display, marker::PhantomData};
+
+use alloc::{boxed::Box, sync::Arc};
+
+use crate::{DynMessage, Message, NoElement, View, ViewId, ViewPathTracker};
+
+/// A `Context` for a [`View`](crate::View) implementation which supports
+/// asynchronous message reporting.
+pub trait AsyncCtx: ViewPathTracker {
+    /// Get a [`Proxy`] for this context.
+    // TODO: Maybe store the current path within this Proxy?
+    fn proxy(&mut self) -> Arc<dyn RawProxy>;
+}
+
+/// A handle to a Xilem driver which can be used to queue a message for a View.
+///
+/// These messages are [`crate::DynMessage`]s, which are sent to a view at
+/// a specific path.
+///
+/// This can be used for asynchronous event handling.
+/// For example, to get the result of a `Future` or a channel into
+/// the view, which then will ultimately.
+///
+/// In the Xilem crate, this will wrap an `EventLoopProxy` from Winit.
+///
+/// ## Lifetimes
+///
+/// It is valid for a [`Proxy`] to outlive the [`View`](crate::View) it is associated with.
+pub trait RawProxy: Send + Sync + 'static {
```

Comment:
I wonder, if we can avoid the `Send` bound, such that this could also be used in xilem_web. I do think we also want some real async in xilem_web (e.g. via web-workers), but I could also imagine, using something like `setInterval` similar as `async_repeat` which AFAIK doesn't need real async, and could e.g. use `web_sys::Event`. (This is probably more relevant to the `DynMessage` type used in the method).